docs/docs/development/hardware-integration/boards-shields-keymaps.md

```diff
@@ -6,47 +6,53 @@ title: Boards, Shields, and Keymaps
 
 The foundational elements needed to get a specific keyboard working with ZMK can be broken down into:
 
-- A [KSCAN driver](https://docs.zephyrproject.org/3.5.0/reference/peripherals/kscan.html), which uses `compatible="zmk,kscan-gpio-matrix"` for GPIO matrix based keyboards, or uses `compatible="zmk,kscan-gpio-direct"` for small direct wires.
-- An optional matrix transform, which defines how the KSCAN row/column events are translated into logical "key positions". This is required for non-rectangular keyboards/matrices, where the key positions don't naturally follow the row/columns from the GPIO matrix.
-- A keymap, which binds each key position to a behavior, e.g. key press, mod-tap, momentary layer, in a set of layers.
+- A [physical layout](physical-layouts.md) that describes the electrical and physical structure of the keyboard, referring to:
+  - A [kscan driver](../../config/kscan.md), which most frequently uses `compatible = "zmk,kscan-gpio-matrix"` for GPIO matrix based keyboards, or `compatible = "zmk,kscan-gpio-direct"` for direct wires.
+  - A [matrix transform](../../config/layout.md), which defines how the kscan row/column events are translated into logical "key positions".
+  - An [optional description](physical-layouts.md#optional-keys-property) of physical key positions and sizes, in order to visualize the keyboard accurately in [ZMK Studio](../../features/studio.md).
+- A [keymap](../../keymaps/index.mdx), which binds each key position to a behavior, e.g. key press, mod-tap, momentary layer, in a set of layers.
+- Other, optional configuration items for features such as encoders or lighting systems.
 
-These three core architectural elements are defined per-keyboard, and _where_ they are defined depends on the specifics of how that
-keyboard works. For an overview on the general concepts of boards and shields, please see the [FAQs on boards and shields](../../faq.md#why-boards-and-shields-why-not-just-keyboard).
+These core architectural elements are defined per-keyboard, and _where_ they are defined depends on the specifics of how that keyboard is built.
+For an overview on the general concepts of boards and shields, also see the [FAQs on boards and shields](../../faq.md#why-boards-and-shields-why-not-just-keyboard).
```

Comment:
I think it might be nice to move that info here and linking here from FAQ, or extracting the relevant text into files and importing it into both. Depends on whether we make hardware integration a top level section.

Reply:
Good point, I'll have a look at pulling that information in here. I think it'd be fine to link to this page from the FAQ in that case.

Reply:
I pulled the board/shield definitions here but left the "why" in the FAQ, seemed to fit better there. I summarized the why in a sentence in the intro.